### ISSUE TITLE
Can't get all headers from WebResourceRequested

### ISSUE THREAD

**Issue by huoyaoyuan** [MicrosoftEdge/WebView2Feedback#1074]
**Description**
I'm using WebResourceRequested like a proxy. However, it doesn't give me enough headers, causing certain request failing.
**Is there any security reason for this? What's the suggested approach then?**

**Version**
SDK: WinUI3-Reunion 0.5-prerelease
Runtime: Evergreen 89.0.774.54
Framework: WinUI
OS: Win10

**Repro Steps**
```csharp
coreWebView2.AddWebResourceRequestedFilter("http*", CoreWebView2WebResourceContext.All);
coreWebView2.WebResourceRequested += async (s, e) =>
{
    // check e.Request.Headers here
}
```

**Screenshots**

Headers I get from DevTools, when WebResourceRequested is turned off:
![image](https://user-images.githubusercontent.com/5644458/111075242-feca0700-8521-11eb-8d83-d3d8c0ba8faf.png)
---
Headers I get from the event using VS debugger:
![image](https://user-images.githubusercontent.com/5644458/111075270-29b45b00-8522-11eb-9e41-ab6a5237a018.png)

There are certain headers missing, specifically Accept-Language.

**Additional context**

[AB#32212995](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/32212995)

**Comment by champnic** [comment]
Hey @huoyaoyuan - thanks for the bug report! I've added it to our backlog to take a look.

**Comment by yildirimcagri** [comment]
Hi - Some headers get added from the network stack after the WebResourceRequested event is fired, but depending on what you are trying to do there may be a workaround. If you need to overwrite the header, you may be able to just set the field via the WebResourceRequested event and it should override the value set by the network stack (AFAIK this should work for all except Cookies field). If you are trying to read the actual header that was sent on the network then you can use the Request object from the WebResourceResponseReceived event. Let us know if this works for your scenario.

**Comment by huoyaoyuan** [comment]
> If you are trying to read the actual header that was sent on the network then you can use the Request object from the WebResourceResponseReceived event. Let us know if this works for your scenario.

It should not work. I need to read the headers *then send the request myself*. The received event doesn't let me do that right?

This can also be done by a custom proxy server, but proxy can't be specified in WinUI yet. Is it the suggested approach?

**Comment by huoyaoyuan** [comment]
@yildirimcagri I have to point out that it's a non-solution for me to read the headers **was sent** - I need the actual headers **would be sent**, and don't let the network stack in browser send it.

If there is any security reason for not allowing this, please clarify. Otherwise, please keep this feature request open.

**Comment by champnic** [comment]
@huoyaoyuan We respect the system proxy settings. Would that work for you, or do you need it to be specific to the app?

**Comment by huoyaoyuan** [comment]
@champnic I need to be specify it per app. For WPF, I can pass arguments when creating WebView2 process, but for WinUI3 such interface isn't exposed yet.

**Comment by huoyaoyuan** [comment]
It is somehow achievable by setting environment variable WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS.

**Comment by champnic** [comment]
Yes - for WInUI3 you should be able to use the environment variables before initializing the WebView2 to specify arguments.
